### ISSUE TITLE
Suggestion: Improve the generation preview

### ISSUE THREAD

**Issue by xvwyh** [lllyasviel/FramePack#239]
I'm talking about the "Next Latents" block.

1. Fix the wrongly clamped colors:
   In hunyuan.py in the function `vae_decode_fake`:
   ```diff
   - images = images.clamp(0.0, 1.0)
   + images = images.clamp(-1.0, 1.0)
   ```
   In demo_gradio.py:
   ```diff
   - preview = (preview * 255.0).detach().cpu().numpy().clip(0, 255).astype(np.uint8)
   + preview = (preview * 127.5 + 127.5).detach().cpu().numpy().clip(0, 255).astype(np.uint8)
   ```

2. Output the preview as an animation instead of a frame strip, as gleaming what motion is about to be generated from a frame strip can be difficult. Use APNG for this, for example, it's perfect for the job considering the small size of the preview, and MP4 would probably wreck it with compression artifacts.
   Won't share the code for this, as my knowledge of python is near non-existent to write it myself, and the code ChatGPT generated is, while functional, completely deranged and full of temporary files, wasting hard drive space and probably not even closing handles.

**Comment by yar85** [comment]
I’m not very proficient in Python programming either, but I still gave it a try 😸
To utils.py:
```
def tensor_to_image_frames(tensor: torch.Tensor, scale: float = 1.0) -> list[Image.Image]:
    np_array = (tensor * 127.5 + 127.5).clamp(0, 255).to(torch.uint8).cpu().numpy()

    if np_array.ndim == 5:          # (B, C, T, H, W) → (B, T, H, W, C)
        np_array = np.transpose(np_array, (0, 2, 3, 4, 1))
        np_array = np_array[0]      # (B, T, H, W, C) → (T, H, W, C)
    elif np_array.ndim == 4:        # (C, T, H, W) → (T, H, W, C)
        np_array = np.transpose(np_array, (1, 2, 3, 0))
    else:
        raise ValueError(f'Unsupported tensor shape: {np_array.shape}')

    if scale == 1.0:
      return [Image.fromarray(frame) for frame in np_array]
    else:
      # new_size = (int(np_array.shape[3] * scale), int(np_array.shape[2] * scale))
      new_size = (int(np_array.shape[2] * scale), int(np_array.shape[1] * scale))
      frames = []
      for frame in np_array:
          img = Image.fromarray(frame).resize(new_size, Image.BILINEAR)
          frames.append(img)
      return frames

def get_temp_file_path(rel_path: list[str]):
    dir_path = os.path.join(tempfile.gettempdir(), *rel_path[:-1])
    os.makedirs(dir_path, exist_ok=True)
    return os.path.join(dir_path, rel_path[-1])
```
To demo_gradio.py:
<pre>
from diffusers_helper.utils import <b>get_temp_file_path, tensor_to_image_frames,</b> save_bcthw_as_mp4, crop_or_pad_yield_mask, soft_append_bcthw, resize_and_center_crop, state_dict_weighted_merge, state_dict_offset_merge, generate_timestamp
.....
<i># after imports</i>
PREVIEW_PATH  = get_temp_file_path(['gradio', 'next-latents.webp'])
PREVIEW_SCALE = 3.0
.....
<i># after `preview = vae_decode_fake(preview)`</i>
preview_frames = tensor_to_image_frames(preview, PREVIEW_SCALE)
preview_frames[0].save(
    PREVIEW_PATH, format='WEBP', save_all=True, append_images=preview_frames[1:],
    duration=int(1100 / len(preview_frames)), loop=0)
preview = f'{PREVIEW_PATH}?t={int(time.time())}'
.....
preview_image = gr.Image(label="Next Latents"<b>, format='file'</b>, height=200, visible=False)
</pre>

Result (some simple body movements): 

https://github.com/user-attachments/assets/8e009ac4-5c61-434e-9d0a-1adfe2221bcc
